agent/consul/leader.go

```diff
@@ -771,6 +771,9 @@ func (s *Server) runACLReplicator(
 
 		index, exit, err := replicateFunc(ctx, logger, lastRemoteIndex)
 		if exit {
+			metrics.SetGauge([]string{"leader", "replication", metricName, "status"},
```

Comment:
Looking through this code, we might also need to set leader.replication.index to zero here as well, but that doesn't block this PR. Thank you for taking the time to fix this.